MapboxDirections/MBRouteOptions.swift

```diff
@@ -319,6 +331,10 @@ open class RouteOptions: NSObject, NSSecureCoding {
             params.append(URLQueryItem(name: "roundabout_exits", value: String(includesExitRoundaboutManeuver)))
         }
         
+        if let locale = locale, let languageCode = (locale as NSLocale?)?.object(forKey: .languageCode) as? String {
```

Comment:
The language code is only the two-letter ISO 369 code. It doesn’t ever include a region or script code; for that, you need the full locale identifier. Unfortunately, the Directions API currently lacks any sort of fallback mechanism. So this code will fail for Simplified Chinese (`zh-Hant`), yet if we were to pass in the full locale identifier, the code would break for a Spanish speaker in Mexico (`es-MX`) instead.

Until the Directions API gains basic language fallback capability, we’ll need to hard-code a whitelist of locales that the Directions API supports.